ckanext/xloader/plugin.py

```diff
@@ -208,3 +229,50 @@ def get_helpers(self):
             xloader_helpers.xloader_status_description,
             'is_xloader_format': xloader_helpers.is_xloader_format,
         }
+
+
+def _should_remove_unsupported_resource_from_datastore(res_dict_or_obj):
+    if not toolkit.asbool(toolkit.config.get('ckanext.xloader.limit_datastore_tables', False)):
+        return False
+    if isinstance(res_dict_or_obj, Resource):
+        return ((not XLoaderFormats.is_it_an_xloader_format(getattr(res_dict_or_obj, 'format', u''))
+                    or getattr(res_dict_or_obj, 'url_changed', False))
+                and (getattr(res_dict_or_obj, 'url_type') == 'upload'
+                    or getattr(res_dict_or_obj, 'url_type') == '')
+                and (getattr(res_dict_or_obj, 'datastore_active', False) 
+                    or getattr(res_dict_or_obj, 'extras', {}).get('datastore_active', False)))
```

Comment:
this is kind of a hack, but so is passing either a dict or Resource object to the same function.
```suggestion
        res_dict_or_obj = res_dict_or_obj .__dict__
```